src/main/java/com/frostwire/jlibtorrent/TorrentStats.java

```diff
@@ -0,0 +1,150 @@
+package com.frostwire.jlibtorrent;
+
+import com.frostwire.jlibtorrent.alerts.Alert;
+import com.frostwire.jlibtorrent.alerts.TorrentAlert;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * It will start an alertEvent listener to keep track of the download
+ * and upload Speed (bytes/sec). The granularity selected will depend
+ * on the speed in which the results of the speed of the Alert event
+ * listener is retrieved, which can have a delay in milliseconds.
+ *
+ * @author haperlot
+ */
+
+public final class TorrentStats {
+
+    private final SessionManager sessionManager;
+    private final TorrentHandle torrentHandle;
+    private final long samplingIntervalInMs;
+    private final long maxHistoryInMs;
+    private Queue<Integer> downloadRate = new LinkedList<>();
+    private Queue<Integer> uploadRate = new LinkedList<>();
+    private long tStart; //for collecting sampling interval time
+    private long MAX_SAMPLES;
+
+    public enum Metric {
+        UploadRate,
+        DownloadRate
+    }
+
+    public TorrentStats(final SessionManager sessionManager, TorrentHandle torrentHandle, long samplingIntervalInMs, long maxHistoryInMs) {
+        this.sessionManager = sessionManager;
+        this.torrentHandle = torrentHandle;
+        this.samplingIntervalInMs = samplingIntervalInMs;
+        this.maxHistoryInMs = maxHistoryInMs;
+        this.tStart = System.currentTimeMillis();
+        this.MAX_SAMPLES = maxHistoryInMs / samplingIntervalInMs;
+        startAlertListener();
+    }
+
+    /**
+     * It will start the alert event listener, as soon as the sampling interval
+     * meets the requested time it will save the values on queues. If the queues
+     * reach max size denoted by the maxHistory time, they will poll() the head
+     * element, in our case the oldest inserted.
+     */
+    public void startAlertListener() {
+        sessionManager.addListener(new AlertListener() {
+            @Override
+            public int[] types() {
+                return null;
```

Comment:
@haperlot, this method is used by the listener to determine what alerts should be processed. Im not sure what happens if you return null, perhaps now it will listen to every alert.

think again what you have to do, to EFFICIENTLY filter the only alert you actually need